### ISSUE TITLE
Alternative definition of Chernoff

### ISSUE THREAD

**Issue by hoheinzollern** [math-comp/analysis#1078]
Perhaps this version of Chernoff's bound is more useful:
```coq
Lemma chernoff (X : {RV P >-> R}) (r a : R) : (0 < r)%R ->
  P [set x | X x >= a]%R <= mmt_gen_fun X r * (expR (- (r * a)))%:E.
Proof.
move=> t0.
rewrite /mmt_gen_fun; have -> : expR \o r \o* X =
    (normr \o normr) \o [the {mfun T >-> R} of expR \o r \o* X].
  by apply: funext => t /=; rewrite normr_id ger0_norm ?expR_ge0.
rewrite expRN lee_pdivl_mulr ?expR_gt0//.
rewrite (le_trans _ (markov _ (expR_gt0 (r * a)) _ _ _))//; last first.
  exact: (monoW_in (@ger0_le_norm _)).
rewrite ger0_norm ?expR_ge0// muleC lee_pmul2l// ?lte_fin ?expR_gt0//.
rewrite [X in _ <= P X](_ : _ = [set x | a <= X x]%R)//; apply: eq_set => t/=.
by rewrite ger0_norm ?expR_ge0// lee_fin ler_expR  mulrC ler_pmul2r.
Qed.
```

**Comment by affeldt-aist** [comment]
I think that we do not use it yet in `probability.v` but since you have an application for this variant I think that it is fair to substitute it.

**Comment by affeldt-aist** [comment]
Can you do a small PR for it?

**Comment by affeldt-aist** [comment]
fixed by PR #1140
